### PR TITLE
Refactor scrapers

### PR DESCRIPTION
**Pull request by piotr921**
Refactor czytam.pl scraper, to improve its performance and code quality.

### CODE REVIEW

File: czytamplscraper/src/main/java/com/blackbook/czytamplscraper/controller/ScraperController.java [hunk 1]
```diff
@@ -2,26 +2,29 @@
...
+    public ScraperController(ScraperService scraperService) {
+        this.scraperService = scraperService;
+    }
 
     @RequestMapping(method = RequestMethod.POST)
-    public void postBookDiscounts() {
+    public SimpleResponse postBookDiscounts() {
         log.info("Transaction: POST /api/czytampl-scraper");
         scraperService.saveResultsInDatabase();
+        return new SimpleResponse(HttpStatus.SC_OK, "Czytam.pl scraper results saved in database!");
```
Comment (ghost) [inline czytamplscraper/src/main/java/com/blackbook/czytamplscraper/controller/ScraperController.java:28]: https://docs.spring.io/spring/docs/3.0.x/javadoc-api/org/springframework/http/ResponseEntity.html

File: czytamplscraper/src/main/java/com/blackbook/czytamplscraper/scraper/BookBuilder.java [hunk 2]
```diff
@@ -1,93 +1,98 @@
...
+    private static final String STORE_PAGE = "http://czytam.pl";
 
+    BookDiscountData buildBookDiscountDataObject(Document detailsPage) {
         return BookDiscountData.builder()
                 .bookstoreId(BOOKSTORE_ID)
-                .price(readBookPrice())
-                .bookDiscountDetails(readPromoDetails())
+                .price(readBookPrice(detailsPage))
+                .bookDiscountDetails(readPromoDetails(detailsPage))
                 .bookData(BookData.builder()
```
Comment (ghost) [inline czytamplscraper/src/main/java/com/blackbook/czytamplscraper/scraper/BookBuilder.java:18]: :+1:

File: czytamplscraper/src/main/java/com/blackbook/czytamplscraper/scraper/BookBuilder.java [hunk 3]
```diff
@@ -1,93 +1,98 @@
...
+        } else {
+            return "Unknown";
+        }
     }
 
-    private String readBookGenre() {
+    String readBookGenre(Document detailsPage) {
         String bookGenre = detailsPage.select(".level-2").select(".active").html();
-        return bookGenre.equals("") ? "Unknown" : bookGenre;
+        return bookGenre.equals("") ? null : bookGenre;
```
Comment (ghost) [inline czytamplscraper/src/main/java/com/blackbook/czytamplscraper/scraper/BookBuilder.java:88]: https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#isEmpty--

File: czytamplscraper/src/main/java/com/blackbook/czytamplscraper/service/ScraperService.java [hunk 4]
```diff
@@ -25,7 +29,7 @@ public void saveResultsInDatabase() {
         ClientHttpRequestFactory requestFactory = new BufferingClientHttpRequestFactory(new SimpleClientHttpRequestFactory());
         RestTemplate restTemplate = new RestTemplate(requestFactory);
 
-        Scraper scraper = new Scraper();
+        Scraper scraper = new Scraper(new Connector(), new BookstoreReader(), new PromotionsPageReader(), new BookBuilder());
```
Comment (ghost) [inline czytamplscraper/src/main/java/com/blackbook/czytamplscraper/service/ScraperService.java:32]: spring ?

File: czytamplscraper/src/test/java/com/blackbook/czytamplscraper/controller/ScraperControllerTest.java [hunk 5]
```diff
@@ -1,45 +1,61 @@
...
     @LocalServerPort
     private int port;
 
-    @Rule
-    public WireMockRule wireMockRule = new WireMockRule(options().port(12002));
-
     @Test
-    public void shouldPostCzytamplaScraper() {
+    public void shouldPostBookDiscounts() throws Exception {
+        // Given
```
Comment (ghost) [inline czytamplscraper/src/test/java/com/blackbook/czytamplscraper/controller/ScraperControllerTest.java:32]: gwt :+1:

**Review by ghost — APPROVED** [review]

**Review by Dzionek95 — APPROVED** [review]

**Pull request merged**